### PR TITLE
Handle case where mention is not more related to an existing user

### PR DESCRIPTION
**Pull request by inkhey**
related to #4137 

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done : tested on preprod env

### CODE REVIEW

File: backend/tracim_backend/lib/core/mention.py [hunk 1]
```diff
@@ -164,8 +165,12 @@ def get_receiver_ids(
         else:
             # send to mentioned user
             user_api = UserApi(session=session, config=config, current_user=None)
-            user = user_api.get_one_by_username(recipient)
-            return [user.user_id]
+            try:
+                user = user_api.get_one_by_username(recipient)
+                return [user.user_id]
+            except UserDoesNotExist:
+                logger.warning(cls, "user of username {} could not been found !".format(recipient))
```
Comment (lebouquetin) [inline backend/tracim_backend/lib/core/mention.py:172]: This log wont help to investiguate because it does not explain why there is an error

Comment (inkhey) [reply backend/tracim_backend/lib/core/mention.py]: updated

File: backend/tracim_backend/lib/core/mention.py [hunk 2]
```diff
@@ -164,8 +165,16 @@ def get_receiver_ids(
...
-            user = user_api.get_one_by_username(recipient)
-            return [user.user_id]
+            try:
+                user = user_api.get_one_by_username(recipient)
+                return [user.user_id]
+            except UserDoesNotExist:
+                logger.warning(
+                    cls,
+                    "user of username {} could not been found to obtain receivers list of event {},"
+                    "user may have changed his username.".format(recipient, event.event_id),
```
Comment (inkhey) [inline backend/tracim_backend/lib/core/mention.py:175]:
```suggestion
                    "user of username {} could not been found to obtain receivers list of event {}, "
                    "user may have changed his username.".format(recipient, event.event_id),
```

File: backend/tracim_backend/lib/core/mention.py [hunk 3]
```diff
@@ -164,8 +165,16 @@ def get_receiver_ids(
...
-            user = user_api.get_one_by_username(recipient)
-            return [user.user_id]
+            try:
+                user = user_api.get_one_by_username(recipient)
+                return [user.user_id]
+            except UserDoesNotExist:
+                logger.warning(
+                    cls,
+                    "user of username {} could not been found to obtain receivers list of event {}, "
+                    "user may have changed his username.".format(recipient, event.event_id),
```
Comment (raphj) [inline backend/tracim_backend/lib/core/mention.py:175]:
```suggestion
                    "Could not find user with username {} while obtaining the receiver list of event {}, "
                    "user may have changed their username.".format(recipient, event.event_id),
```

**Review by raphj — APPROVED** [review]

**Pull request merged**